plugins/com.google.cloud.tools.eclipse.dataflow.ui/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/DefaultedPipelineOptionsComponent.java

```diff
@@ -52,6 +51,13 @@
 
   public DefaultedPipelineOptionsComponent(Composite parent, Object layoutData,
       MessageTarget messageTarget, DataflowPreferences preferences) {
+    this(parent, layoutData, messageTarget, preferences, null);
```

Comment:
What do you think of passing `new RunOptionsDefaultsComponent()` instead of `null`? That could simplify the `?` ternay-operator below.

Reply:
I agree it's awkward.  Unfortunately cannot as it's parented using the `defaultsGroup` which is created inside the constructor.  I had explored creating an interface `RunOptionsContainer` to represent the `RunOptionsDefaultsComponent` API, so that it could be more easily mocked, but it didn't much help.

Reply:
Ah, gotcha.